### PR TITLE
[4.0] Disable status module links

### PR DESCRIPTION
**Pull request by brianteeman**
In Joomla 3.x we disabled the user links in the top right hand corner of the admin at the same time that we disabled the main menu. Amongst other reasons this was to prevent people from leaving an item checked out.

In Joomla 4 we have now have several icons in this place (users, private-messages, post-installation, multilingual status, preview) and these are from the status module and the multilingual status module.

Nothing needs to be changed with the multilingual status module as this opens in a model and preview is also ok because it opens a new tab.

But users, private-messages, post-installation all need to be disabled when an item is open to prevent the item being left checked out.

In J3 and in J4 we have code for this ```$app->input->get('hidemainmenu');```

In J4 the variable is declared but never used - whoops :)

This PR fixes that.

### Test
Open an article and ensure that only the "preview" and if enabled "multilingual status" links work
Close the article and ensure that all of them work.

### Note
The template design team might want to change the icon or styling when it is disabled but that's beyond the scope of this PR.

### CODE REVIEW

File: administrator/modules/mod_status/tmpl/default.php [hunk 1]
```diff
@@ -44,7 +44,7 @@
 		</li>
 
 		<li class="nav-item">
-			<a class="nav-link dropdown-toggle" href="<?php echo Route::_('index.php?option=com_messages'); ?>" title="<?php echo Text::_('MOD_STATUS_PRIVATE_MESSAGES'); ?>">
+			<a class="nav-link <?php echo ($hideLinks ? ' disabled' : 'dropdown-toggle'); ?>" <?php echo ($hideLinks ? '' : 'href="' . Route::_('index.php?option=com_messages') . '"'); ?> title="<?php echo Text::_('MOD_STATUS_PRIVATE_MESSAGES'); ?>">
```
Comment (Quy) [inline administrator/modules/mod_status/tmpl/default.php]:
```suggestion
			<a class="nav-link <?php echo ($hideLinks ? 'disabled' : 'dropdown-toggle'); ?>" <?php echo ($hideLinks ? '' : 'href="' . Route::_('index.php?option=com_messages') . '"'); ?> title="<?php echo Text::_('MOD_STATUS_PRIVATE_MESSAGES'); ?>">
```

File: administrator/modules/mod_status/tmpl/default.php [hunk 2]
```diff
@@ -56,49 +56,53 @@
 
 		<?php if ($user->authorise('core.manage', 'com_postinstall')) : ?>
 		<li class="nav-item dropdown">
-			<a class="nav-link dropdown-toggle" href="#" data-toggle="dropdown" title="<?php echo Text::_('MOD_STATUS_POST_INSTALLATION_MESSAGES'); ?>">
+			<a class="nav-link <?php echo ($hideLinks ? ' disabled' : 'dropdown-toggle'); ?>" data-toggle="<?php echo ($hideLinks ? '' : 'dropdown'); ?>" <?php echo ($hideLinks ? '' : 'href="#"'); ?> title="<?php echo Text::_('MOD_STATUS_POST_INSTALLATION_MESSAGES'); ?>">
```
Comment (Quy) [inline administrator/modules/mod_status/tmpl/default.php]:
```suggestion
			<a class="nav-link <?php echo ($hideLinks ? 'disabled' : 'dropdown-toggle'); ?>" data-toggle="<?php echo ($hideLinks ? '' : 'dropdown'); ?>" <?php echo ($hideLinks ? '' : 'href="#"'); ?> title="<?php echo Text::_('MOD_STATUS_POST_INSTALLATION_MESSAGES'); ?>">
```

File: administrator/modules/mod_status/tmpl/default.php [hunk 3]
```diff
@@ -56,49 +56,53 @@
...
+					</div>
 				</div>
-			</div>
+			<?php endif; ?>
 		</li>
 		<?php endif; ?>
 
 		<li class="nav-item dropdown header-profile">
-			<a class="nav-link dropdown-toggle" href="#" data-toggle="dropdown" title="<?php echo Text::_('MOD_STATUS_USER_MENU'); ?>">
+			<a class="nav-link  <?php echo ($hideLinks ? ' disabled' : 'dropdown-toggle'); ?>" data-toggle="<?php echo ($hideLinks ? '' : 'dropdown'); ?>" <?php echo ($hideLinks ? '' : 'href="#"'); ?> title="<?php echo Text::_('MOD_STATUS_USER_MENU'); ?>">
```
Comment (Quy) [inline administrator/modules/mod_status/tmpl/default.php]:
```suggestion
			<a class="nav-link <?php echo ($hideLinks ? 'disabled' : 'dropdown-toggle'); ?>" data-toggle="<?php echo ($hideLinks ? '' : 'dropdown'); ?>" <?php echo ($hideLinks ? '' : 'href="#"'); ?> title="<?php echo Text::_('MOD_STATUS_USER_MENU'); ?>">
```

File: administrator/modules/mod_status/tmpl/default.php [hunk 4]
```diff
@@ -56,49 +56,53 @@
 
 		<?php if ($user->authorise('core.manage', 'com_postinstall')) : ?>
 		<li class="nav-item dropdown">
-			<a class="nav-link dropdown-toggle" href="#" data-toggle="dropdown" title="<?php echo Text::_('MOD_STATUS_POST_INSTALLATION_MESSAGES'); ?>">
+			<a class="nav-link <?php echo ($hideLinks ? 'disabled' : 'dropdown-toggle'); ?>" data-toggle="<?php echo ($hideLinks ? '' : 'dropdown'); ?>" <?php echo ($hideLinks ? '' : 'href="#"'); ?> title="<?php echo Text::_('MOD_STATUS_POST_INSTALLATION_MESSAGES'); ?>">
```
Comment (Quy) [inline administrator/modules/mod_status/tmpl/default.php]:
```suggestion
			<a class="nav-link <?php echo ($hideLinks ? 'disabled' : 'dropdown-toggle'); ?>" <?php echo ($hideLinks ? '' : 'data-toggle="dropdown" href="#"'); ?> title="<?php echo Text::_('MOD_STATUS_POST_INSTALLATION_MESSAGES'); ?>">
```

File: administrator/modules/mod_status/tmpl/default.php [hunk 5]
```diff
@@ -56,49 +56,53 @@
...
+					</div>
 				</div>
-			</div>
+			<?php endif; ?>
 		</li>
 		<?php endif; ?>
 
 		<li class="nav-item dropdown header-profile">
-			<a class="nav-link dropdown-toggle" href="#" data-toggle="dropdown" title="<?php echo Text::_('MOD_STATUS_USER_MENU'); ?>">
+			<a class="nav-link <?php echo ($hideLinks ? 'disabled' : 'dropdown-toggle'); ?>" data-toggle="<?php echo ($hideLinks ? '' : 'dropdown'); ?>" <?php echo ($hideLinks ? '' : 'href="#"'); ?> title="<?php echo Text::_('MOD_STATUS_USER_MENU'); ?>">
```
Comment (Quy) [inline administrator/modules/mod_status/tmpl/default.php]:
```suggestion
			<a class="nav-link <?php echo ($hideLinks ? 'disabled' : 'dropdown-toggle'); ?>" <?php echo ($hideLinks ? '' : 'data-toggle="dropdown" href="#"'); ?> title="<?php echo Text::_('MOD_STATUS_USER_MENU'); ?>">
```

**Comment by brianteeman** [comment]
Thanks @quy

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 457b3ef468082ee4591cda6540889b5c7f0f3f77<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23752">issues.joomla.org/tracker/joomla-cms/23752</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Comment by brianteeman** [comment]
Thanks

**Pull request merged**